stack.yaml

```diff
@@ -131,3 +142,11 @@ ghc-options:
   cardano-sl:            -Werror
   cardano-sl-tools:      -Werror
   cardano-sl-lwallet:    -Werror
+  cardano-sl-explorer:   -Werror
+
+# Added for explorer
+flags:
+  cardano-sl:
+    with-explorer: True
```

Comment:
1. We want to be able to compile with explorer and without explorer.
2. Building `cardano-sl-explorer` without passing `with-explorer` flag to `cardano-sl` is wrong, don't do it. Build script should detect whether we want to build explorer and enable `with-explorer` flag if we do and disable it if we don't. Regarding CI: I don't know which approach will be used, it should be discussed with devops, ideally we should build both version.
3. `with-explorer` flag is a dirty and annoying hack which shouldn't exist, but nobody knows when we will find time to remove it. Maybe it's time to do it.
4. Regarding default value: personally I don't really care as long as `build/cardano-sl.sh` has the desired behavior. For example, if I use `scripts/build/cardano-sl.sh sl` it should build without explorer (as it's done now). If I use `scripts/build/cardano-sl.sh` without any arguments, it means that I want to build everything, so it should figure out that I want explorer and `with-explorer` should be enabled.

P. S. Gold linker issues are irrelevant, they are already solved. If you build explorer you must enable `with-explorer` for `cardano-sl`. In this case everything should work.